spec/src/main/asciidoc/batch_runtime_spec.adoc

```diff
@@ -63,32 +63,97 @@ See section xref:batchruntime[10.9.5] for details on the BatchRuntime class.
 
 === Batch Artifact Loading
 
-All batch artifacts comprising a batch application are loadable by the
-following loaders in the order specified:
+By "artifact loading" we mean the process by which an instance of a batch API 
+interface or class is obtained by the batch runtime during the execution of the job, from
+the job definition.
 
-1.  Implementation-specific loader +
-+
-The batch runtime implementation _may_ provide an
-implementation-specific means by which batch artifacts references in a Job XML (i.e. via the 'ref=' attribute) are resolved to an implementation class and instantiated. When the batch runtime resolves a batch artifact reference to an instance the implementation-specific mechanism (if one exists) is attempted first. The loader must return an
-instance or null. +
+In other words, this describes how a reference in a Job XML (using the 'ref' attribute) 
+is resolved to an instance of an implementation class.
+
+==== Required Artifact Loading Sequence
+
+Batch artifacts must be loaded in the order specified:
+
+1. CDI Bean - using 'ref' as EL name
 +
-An example of an implementation-specific loader might be CDI or Spring DI.
-2.  Archive loader +
+The batch runtime must attempt to pass the 'ref' value as an EL bean name to the
+current CDI context, (e.g. using `BeanManager.getBeans(String name)` to obtain a CDI Bean instance, 
+and then obtain a contextual reference for this bean for use by the batch runtime.
+
+2. CDI Bean - using batch.xml mapping to fully qualified class name (FQCN)
```

Comment:
think 2 and 3 should be merged, archive loader is not a thing users see nor the spec defines by itself (it is more an implementation details) so it is really loadClass(ref) - whatever classloader(s) it is.

Reply:
I had the same reaction reading this.   I don't especially like the "archive loader" wording and wouldn't have thought to word it like that.   But TCCL does not necessarily == the classloader loading the batch.xml, so I think it needs a bit more work to remain true to the intention behind the earlier wording.

Reply:
Made a 2nd commit.  I think this new wording works.